### PR TITLE
Some tensor mechanics test requirements, issues, and design page links

### PR DESCRIPTION
**Pull request by sapitts**
Includes the sqa documentation for the design and requirements for a small set of isotropic elastic tensor mechanics tests.

Refs #9638

### CODE REVIEW

**Comment by sapitts** [comment]
pinging @aeslaughter and @bwspenc for review please

File: modules/tensor_mechanics/test/tests/1D_spherical/tests [hunk 1]
```diff
@@ -6,6 +6,9 @@
     compiler = 'CLANG GCC'
     cli_args = 'Modules/TensorMechanics/Master/all/incremental=false
                 Materials/stress/type=ComputeLinearElasticStress'
+    issues = '#6256 7555'
+    design = 'source/Materials/ComputeRSphericalSmallStrain.md source/Materials/stress/type=ComputeLinearElasticStress.md syntax/Modules/TensorMechanics/Master/index.md'
+    requirement = 'The ComputeRSphericalSmallStrain class, called through the TensorMechanicsMaster action, shall find the total linearized solution for the displacement of a solid isotropic elastic sphere under an applied load on the outer radius of the sphere of -1MPa as $u(r) = \frac{- P * (1 - 2 * v) * r}{E}$ where P is the applied pressure, b is the outer radius, a is the inner radius, v is Poisson's ration, E is Young's Modulus, and r is the radial position.'
```
Comment (bwspenc) [inline modules/tensor_mechanics/test/tests/1D_spherical/tests]: This seems to me like we're getting a little too specific with the equation. I'd suggest something like this:

The mechanics system shall be capable of accurately computing the displacement of a thick-walled sphere with a pressure applied to the outer surface using a 1D spherical symmetric representation with elastic, total small strain assumptions.

File: modules/tensor_mechanics/test/tests/anisotropic_patch/tests [hunk 2]
```diff
@@ -4,5 +4,8 @@
     input = 'anisotropic_patch_test.i'
     exodiff = 'anisotropic_patch_test_out.e'
     scale_refine = 1
+    issues = '#4716 #7555'
+    design = 'source/Materials/ComputeLinearElasticStress.md source/Materials/ComputeSmallStrain.md syntax/Modules/TensorMechanics/Master/index.md'
+    requirement = 'The ComputeLinearElasticStress class shall compute and output the xx, yy, zz, xy, yz, and zx stress components on an irregular hex mesh compused of seven elements which form a unit cube with one internal element.'
```
Comment (bwspenc) [inline modules/tensor_mechanics/test/tests/anisotropic_patch/tests]: That comment in the .i file doesn't really say what this is testing.  I'd go with something like this:

The mechanics system shall be capable of accurately computing the elastic response of an anisotropic elastic material where 9 components of a symmetric elasticity tensor are specified on an irregular patch of elements with total small strain assumptions.

File: modules/tensor_mechanics/test/tests/finite_strain_elastic/tests [hunk 3]
```diff
@@ -3,53 +3,80 @@
     type = 'Exodiff'
     input = 'finite_strain_elastic_new_test.i'
     exodiff = 'finite_strain_elastic_new_test_out.e'
+    issues = '#4716 #7555'
+    design = 'source/Materials/ComputeFiniteStrainElasticStress.md source/Materials/ComputeFiniteStrain.md syntax/Modules/TensorMechanics/Master/index.md'
+    reuirement = 'The ComputeFiniteStrainElasticStress class shall compute the elastic stress for a finite strain formulation found with the Taylor expansion from Rashid, Numerical Methods in Engineering (1993) on a unit 3D cube in a Cartesian system using the TensorMechanicsMaster action.'
```
Comment (bwspenc) [inline modules/tensor_mechanics/test/tests/finite_strain_elastic/tests]: I think I'd  just refer to the reference as 'Rashid (1993)' to keep it shorter.

File: modules/tensor_mechanics/test/tests/1D_spherical/tests [hunk 4]
```diff
@@ -15,11 +18,17 @@
     cli_args = 'Modules/TensorMechanics/Master/all/incremental=true
                 Materials/stress/type=ComputeFiniteStrainElasticStress'
     prereq = 'smallStrain_1DSphere'
+    issues = '#6256 7555'
+    design = 'source/Materials/ComputeRSphericalIncrementalStrain.md source/Materials/stress/type=ComputeFiniteStrainElasticStress.md syntax/Modules/TensorMechanics/Master/index.md'
+    requirement = 'The ComputeRSphericalIncrementalStrain class, called through the TensorMechanicsMaster action, shall find the linearized incremental strain displacement of a solid isotropic elastic sphere under an applied load on the outer radius of the sphere of -1MPa as $u(r) = \frac{- P * (1 - 2 * v) * r}{E}$ where P is the applied pressure, b is the outer radius, a is the inner radius, v is Poisson's ration, E is Young's Modulus, and r is the radial position.'
```
Comment (bwspenc) [inline modules/tensor_mechanics/test/tests/1D_spherical/tests]: similar deal on these other ones

File: modules/tensor_mechanics/test/tests/finite_strain_elastic/tests [hunk 5]
```diff
@@ -3,53 +3,80 @@
...
+    design = 'source/Materials/ComputeMultiPlasticityStress tensor_mechanics/VolumetricLocking.md'
+    requirement = 'The ComputeMultiPlasticityStress class shall, when supplied with no plastic models, reduce to and produce the solely elastic stress solution for a finite strain fomulation, using the volumetric locking correction b-bar formulation.'
   [../]
   [./rotation_new]
     type = 'Exodiff'
     input = 'elastic_rotation_test.i'
     exodiff = 'elastic_rotation_test_out.e'
+    issues = '#4716 #7555'
+    design = 'source/Materials/ComputeFiniteStrainElasticStress.md syntax/Modules/TensorMechanics/Master/index.md'
+    requirement = 'The ComputeFiniteStrainElasticStress class shall compute the elastic stress based on a finite strain fomulation and then follow the stress as the mesh is rotated 90 degrees in accordance with Kamojjala et al. Engineering with Computers (31) 2015 using the TensorMechanicsMaster action.'
```
Comment (bwspenc) [inline modules/tensor_mechanics/test/tests/finite_strain_elastic/tests]: Shorten the reference to 'Kamojjala et al. (2015)'

File: modules/tensor_mechanics/test/tests/finite_strain_elastic/tests [hunk 6]
```diff
@@ -3,53 +3,80 @@
...
   [../]
   [./rotation_new_Bbar]
     type = 'Exodiff'
     input = 'elastic_rotation_test.i'
     exodiff = 'elastic_rotation_test_out.e'
     cli_args = 'GlobalParams/volumetric_locking_correction=true'
     prereq = 'rotation_new'
+    issues = '#4716 #8235'
+    design = 'source/Materials/ComputeFiniteStrainElasticStress.md tensor_mechanics/VolumetricLocking.md'
+    requirement = 'The ComputeFiniteStrainElasticStress class shall compute the elastic stress based on a finite strain fomulation and then follow the stress as the mesh is rotated 90 degrees in accordance with Kamojjala et al. Engineering with Computers (31) 2015 using the volumetric locking correction b-bar formulation.'
```
Comment (bwspenc) [inline modules/tensor_mechanics/test/tests/finite_strain_elastic/tests]: same deal with the reference

File: modules/tensor_mechanics/test/tests/finite_strain_elastic/tests [hunk 7]
```diff
@@ -3,53 +3,80 @@
...
   [../]
   [./eigen_sol]
     type = 'Exodiff'
     input = 'finite_strain_elastic_eigen_sol.i'
     exodiff = 'finite_strain_elastic_eigen_sol_out.e'
+    issues = '#4716 #7555'
+    design = 'source/Materials/ComputeFiniteStrainElasticStress.md source/Materials/ComputeFiniteStrain.md syntax/Modules/TensorMechanics/Master/index.md'
+    reuirement = 'The ComputeFiniteStrainElasticStress class shall compute the elastic stress for a finite strain formulation found with the eigen-strain solution on a unit 3D cube in a Cartesian system using the TensorMechanicsMaster action.'
```
Comment (bwspenc) [inline modules/tensor_mechanics/test/tests/finite_strain_elastic/tests]: 'found with the eigen-strain solution' -> 'using a direct eigensolution to perform the polar decomposition of stretch and rotation'

File: modules/tensor_mechanics/test/tests/finite_strain_elastic/tests [hunk 8]
```diff
@@ -3,53 +3,80 @@
...
   [../]
   [./eigen_sol_Bbar]
     type = 'Exodiff'
     input = 'finite_strain_elastic_eigen_sol.i'
     exodiff = 'finite_strain_elastic_eigen_sol_out.e'
     cli_args = 'GlobalParams/volumetric_locking_correction=true'
     prereq = 'eigen_sol'
+    issues = '#8235 #4716'
+    design = 'source/Materials/ComputeFiniteStrainElasticStress.md source/Materials/ComputeFiniteStrain.md tensor_mechanics/VolumetricLocking.md'
+    requirement = 'The ComputeFiniteStrainElasticStress class shall compute the elastic stress for a finite strain formulation found with the eigen-strain solution on a unit 3D cube in a Cartesian system using the volumetric locking correction b-bar formulation.'
```
Comment (bwspenc) [inline modules/tensor_mechanics/test/tests/finite_strain_elastic/tests]: same deal with the eigen-strain text replacement

File: modules/tensor_mechanics/test/tests/finite_strain_elastic/tests [hunk 9]
```diff
@@ -3,53 +3,80 @@
...
   [../]
   [./stress_errorcheck]
     type = 'RunException'
     input = 'finite_strain_stress_errorcheck.i'
     expect_err = 'This linear elastic stress calculation only works for small strains; use ComputeFiniteStrainElasticStress for simulations using incremental and finite strains.'
+    issues = '#4716 #7555'
+    design = 'source/Materials/ComputeFiniteStrainElasticStress.md source/Materials/ComputeLinearElasticStress.md syntax/Modules/TensorMechanics/Master/index.md'
+    requirement = 'The ComputeLinearElasticStress class shall error if a user attempts to run a problem using ComputeLinearElasticStress with a finite strain formulation.'
```
Comment (bwspenc) [inline modules/tensor_mechanics/test/tests/finite_strain_elastic/tests]: The ComputeLinearElasticStress class shall generate an error if a user attempts to run a problem using ComputeLinearElasticStress with a finite strain formulation.

File: modules/tensor_mechanics/test/tests/pressure/tests [hunk 10]
```diff
@@ -3,12 +3,18 @@
     type = Exodiff
     input = 'pressure_test.i'
     exodiff = 'pressure_test_out.e'
+    issues = '#4781'
+    design = 'syntax/BCs/Pressure/index.md'
+    requirement = 'The pressure boundary condition action shall compute pressure loads on three faces of an elastic small strain isotropic unit cube to create a hydrostatic pressure which corresponds to the values of the displacements.'
```
Comment (bwspenc) [inline modules/tensor_mechanics/test/tests/pressure/tests]: The Pressure boundary condition action shall create the objects needed to apply pressure boundary conditions on a 3D model as demonstrated by correctly computing the response of an elastic small-strain isotropic unit cube with pressure applied on three faces to create a hydrostatic pressure.

File: modules/tensor_mechanics/test/tests/pressure/tests [hunk 11]
```diff
@@ -3,12 +3,18 @@
...
   [../]
   [./3D_Bbar]
     type = Exodiff
     input = 'pressure_test.i'
     exodiff = 'pressure_test_out.e'
     cli_args = 'GlobalParams/volumetric_locking_correction=true'
     prereq = '3D'
+    issues = '#4781 #8235'
+    design = 'syntax/BCs/Pressure/index.md'
+    requirement = 'The pressure boundary condition action shall compute pressure loads on three faces of an elastic small strain isotropic unit cube to create a hydrostatic pressure which corresponds to the values of the displacements using the volumetric locking correction b-bar formulation.'
```
Comment (bwspenc) [inline modules/tensor_mechanics/test/tests/pressure/tests]: The Pressure boundary condition action shall create the objects needed to apply pressure boundary conditions on a 3D model as demonstrated by correctly computing the response of an elastic small-strain isotropic unit cube with pressure applied on three faces to create a hydrostatic pressure using the volumetric locking correction b-bar formulation.

**Review by bwspenc — CHANGES_REQUESTED** [review]

File: modules/tensor_mechanics/test/tests/pressure/pressure_test.i [hunk 12]
```diff
@@ -10,7 +10,6 @@
 
 
 [Mesh]
-  # Comment
```
Comment (bwspenc) [inline modules/tensor_mechanics/test/tests/pressure/pressure_test.i]: 👍

**Review by bwspenc — DISMISSED** [review]

File: modules/tensor_mechanics/test/tests/1D_spherical/tests [hunk 13]
```diff
@@ -6,6 +6,9 @@
     compiler = 'CLANG GCC'
     cli_args = 'Modules/TensorMechanics/Master/all/incremental=false
                 Materials/stress/type=ComputeLinearElasticStress'
+    issues = '#6256 7555'
```
[the remark this answers is not in the document]
Comment (sapitts) [reply modules/tensor_mechanics/test/tests/1D_spherical/tests]: obviously it's c ;-) thanks for catching this

**Review by bwspenc — APPROVED** [review]

**Pull request merged**